rakelib/pkg.rake

```diff
@@ -0,0 +1,67 @@
+CLOBBER.include("pkg/**/*.gem")
+CLEAN.include("tmp/pkg")
+CLEAN.include("tmp/pkg")
+
+namespace :pkg do
+  directory "pkg"
+
+  desc "Build the source gem (#{GEMSPEC.name}-#{GEMSPEC.version}.gem)"
+  task ruby: "pkg" do
+    slug = "#{GEMSPEC.name}-#{GEMSPEC.version}"
+    output_gempath = File.expand_path("../pkg/#{slug}.gem", __dir__)
+    gemspec_path = "wasmtime.gemspec"
+    base_dir = File.join("tmp/pkg", slug)
+    staging_dir = File.join(base_dir, "stage")
+    unpacked_dir = File.join(base_dir, "unpacked")
+    vendor_dir = "ext/vendor"
+    staging_gem_path = File.join(staging_dir, "#{slug}.gem")
+
+    puts "Building source gem..."
+
+    rm(output_gempath) if File.exist?(output_gempath)
+    rm_rf(staging_dir)
+    rm_rf(unpacked_dir)
+    mkdir_p(staging_dir)
+    cp(gemspec_path, staging_dir)
+
+    GEMSPEC.files.each do |file|
+      dest = File.join(staging_dir, file)
+      mkdir_p(File.dirname(dest))
+      cp(file, dest) if File.file?(file)
+    end
+
+    Dir.chdir(staging_dir) do
+      cargo_config_path = ".cargo/config"
+      final_gemspec = Gem::Specification.load(File.basename(gemspec_path))
+
+      puts "Vendoring cargo dependencies to #{cargo_config_path}..."
+      mkdir_p ".cargo"
+      sh "cargo vendor --versioned-dirs --locked #{vendor_dir} >> #{cargo_config_path} 2>/dev/null"
+
+      final_gemspec.files += dirglob("**/.cargo/**/*")
+      final_gemspec.files += dirglob("./#{vendor_dir}/**/*")
+      final_gemspec.files.reject! { |f| File.directory?(f) }
```

Comment:
Nit: first time I ran this, it didn't complete because I had empty dirs from some prior exploration and that seemed to break the script. Somehow the gemspec ends up including files that do not exist, not sure why:

```
Gem::InvalidSpecificationException: ["ext/wasmtime"] are not files
wasmtime-rb/rakelib/pkg.rake:46:in `block (3 levels) in <top (required)>'
```

Removing the empty dir fixed it for me.  If you want to repro: `mkdir -p ext/wasmtime && rake pkg:ruby`

Reply:
solved in https://github.com/bytecodealliance/wasmtime-rb/pull/49/commits/08ef238cce08418af9853d467f3947ea2ae2f16a